src/adaPets.js

```diff
@@ -9,27 +9,53 @@ const BASE_URL = 'http://localhost:3000/pets/';
 
 // Option functions.
 const listPets = () => {
-  // Fill out as part of Wave 1.
+  axios.get(BASE_URL)
+    .then((response) => {
+      setResult(response.data);
+    })
+    .catch((error) => {
+      setError(error.message);
+    });
 };
 
 const showDetails = (selectedPetId) => {
   if (!selectedPetId) {
     setError("You tried to show details for a pet without selecting it!");
   } else {
-    // Fill out as part of Wave 2.
+    axios.get(`${BASE_URL + selectedPetId}`)
+      .then((response) => {
+        setResult(response.data);
+      })
+      .catch((error) => {
+        setError(error.message);
+      });
   }
 };
 
 const removePet = (selectedPetId) => {
   if (!selectedPetId) {
     setError("You tried to remove a pet without selecting it!");
   } else {
-    // Fill out as part of Wave 3.
+    axios.delete(`${BASE_URL + selectedPetId}`)
+      .then((response) => {
+        setResult(response.data);
+      })
+      .catch((error) => {
+        setError(error.message);
```

Comment:
A test for this error condition is failing because it is looking to match the word "remove".  Consider something like this:
```suggestion
        setError(`Failed to remove a pet with status code: ${error.response.status}`);
```